drivers/ethernet/eth_stm32_hal.c

```diff
@@ -55,6 +55,7 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #endif
 
 #define PHY_ADDR	CONFIG_ETH_STM32_HAL_PHY_ADDRESS
+#define ETH_INIT_SPLIT  89
```

Comment:
Please use KConfig functionality for it.

Reply:
Lie above - using Kconfig deinition, like for example:
config ETH_STM32_HAL_SPLIT_INIT_PRIO generates weird linker error like:
/../arm-none-eabi/bin/ld.bfd: Undefined initialization levels used.
The same situation like with symbolic expression :|

Reply:
This is really strange, but I have added this snipped to Kconfig:
`config ETH_STM32_HAL_SPLIT_INIT_PRIO
	int
	default 89
	help
		Ethernet initalization procedure splitted on two sections`

But...the same result....
`/bin/ld.bfd: Undefined initialization levels used.`

Reply:
Thanks for changes and PR done. It looks better and better now.
Could you please push your KConfig changes too, s.t. I can see?

Reply:
Oki, nice to see. So the last one (amended) commit shows the KConfig modification and the `INIT()` macro which is using it.

Reply:
@KozhinovAlexander - the pipeline will failed because of this change, but I just want to show you my KConfig file